src/coreclr/jit/fgbasic.cpp

```diff
@@ -999,28 +1097,145 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, Fixed
                     isJitIntrinsic = eeIsJitIntrinsic(methodHnd);
                 }
 
-                if (isJitIntrinsic)
+                if (isJitIntrinsic && conservativeInliner)
```

Comment:
Consider moving some of this logic to helper methods; this method is already huge.